### PR TITLE
AggregateType TryGetPrimaryKeyFieldName return an enumerable and add PrimaryKeyNames to TableDelegationInfo

### PR DESCRIPTION
**Pull request by LucGenetier**
Make AggregateType TryGetPrimaryKeyFieldName return an enumerable to allow composed keys
Add PrimaryKeyNames to TableDelegationInfo

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs [hunk 1]
```diff
@@ -64,6 +65,10 @@ public RecordType(DisplayNameProvider displayNameProvider, TableDelegationInfo d
 
         public override IEnumerable<string> FieldNames => _fieldNames;
 
+        // List of fields names that compose the primary key
+        // This array is ordered and contains logical names
+        public virtual string[] PrimaryKeyNames => Array.Empty<string>();
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs:70]: >PrimaryKeyNames [](http://example.com/codeflow?start=32&length=15)

It looks like we have something similar already, can we reuse this: 

[https://github.com/microsoft/Power-Fx/blob/46a1bfc11a173fd22c04f3b0b2ae7a55b72e6368/src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs#L203](https://github.com/microsoft/Power-Fx/blob/46a1bfc11a173fd22c04f3b0b2ae7a55b72e6368/src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs#L203)

File: src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs [hunk 2]
```diff
@@ -52,6 +52,10 @@ public abstract class TableDelegationInfo
         // Dataset name
         public string DatasetName { get; init; }
 
+        // Supports primary key names (multiple when composed key)
+        // This array is ordered
+        public string[] PrimaryKeyNames { get; init; }
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs:57]: >string[] [](http://example.com/codeflow?start=15&length=8)

nit - arrays are mutable. 
Can we expose this as something immutable  like IReadOnlyCollection<string> ?

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs (hunk 1 above)
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs:70]: I'll move that one to AggregateType
This one is different in the sense that we need to support composed keys

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by MikeStall** [comment]
>         [Obsolete("preview")]

Note this method is Obsolete(Preview), so we may be able to make a breaking change to it. 

Having redundant virtuals is a maintenance concern - do callers need to override both together? 


---
Refers to: src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs:202 in a15cc23. [](commit_id = a15cc23d1c0bba4302ee04643910a4318bb0fdf3, deletion_comment = False)

File: src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+using System;
```
Comment (anderson-joyle) [inline src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs:4]: nit: needed?

Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs]: It's not, I'll remove it

**Comment by LucGenetier** [comment]
>         [Obsolete("preview")]

I'll make that one return an enumerable to manage composed key

---
In reply to: [2523912736](https://github.com/microsoft/Power-Fx/pull/2770#issuecomment-2523912736) [](http://example.com/codeflow?ancestors=2523912736)

---
Refers to: src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs:202 in a15cc23. [](commit_id = a15cc23d1c0bba4302ee04643910a4318bb0fdf3, deletion_comment = False)

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Types.AggregateType.TryGetPrimaryKeyFieldName(System.String@)

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Types.AggregateType.TryGetPrimaryKeyFieldName(System.String@)

**Comment by MikeStall** [comment]
>         [Obsolete("preview")]

tagged PR with "Breaking API" 

---
In reply to: [2527893404](https://github.com/microsoft/Power-Fx/pull/2770#issuecomment-2527893404) [](http://example.com/codeflow?ancestors=2527893404,2523912736)

---
Refers to: src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs:202 in a15cc23. [](commit_id = a15cc23d1c0bba4302ee04643910a4318bb0fdf3, deletion_comment = False)

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Types.AggregateType.TryGetPrimaryKeyFieldName(System.String@)

**Pull request merged**